graphql_compiler/interpreter/expression_ops/immediate_expression_handlers.py

```diff
@@ -23,10 +23,13 @@ def evaluate_local_field(
     expression: LocalField,
     data_contexts: Iterable[DataContext],
 ) -> Iterator[Tuple[DataContext, Any]]:
-    # TODO(predrag): Add hints here.
+    vertex_location = expression.location.at_vertex()
```

Comment:
I knew there was a reason I didn't add hints to `project_property` the first time around 😄

Reply:
We could just convert local fields to context fields as an IR pass, but I think that's problematic, since the context consists of locations already processed.

The alternative is to plumb the current location in here through the dispatcher, and everywhere else that's required to make that work.

Reply:
It's also illegal to do that conversion — `LocalField` expressions show up _before_ the location is declared in a `MarkLocation` block, so they'd be forward-referencing a still-undeclared location value.

We probably need to do the current location plumbing.